src/syntax/sugar.rkt

```diff
@@ -1,70 +1,67 @@
 ;;
-;; ## FPCore (with a couple modifications) ##
+;; FPCore: the input/output language
 ;;
-;; - input language, loopless, with bindings, lightly annotated
-;; - operators denote real computations
-;; - rounding context (in addition to variable context) decides format at node
+;; - standardized interchange format with other tools
+;; - using loopless, tensorless subset
+;; - operators denote real computations while rounding contexts decide format
 ;;
 ;;  <FPCore> ::= (FPCore (<var> ...) <props> ... <expr>)
-;;           ::= (FPCore <name> (<var> ...) <props> ... <expr>)
-;;  <expr>   ::= (let ([<var> <expr>] ...) <expr>)
-;;           ::= (let* ([<var> <expr>] ...) <expr>)
-;;           ::= (if <expr> <expr> <expr>
-;;           ::= (! <props> ... <expr>)
-;;           ::= (<op> <expr> ...)
-;;           ::= <var>
-;;           ::= <number>
+;;             | (FPCore <id> (<var> ...) <props> ... <expr>)
+;;
+;;  <expr>   ::= (let ([<id> <expr>] ...) <expr>)
+;;             | (let* ([<id> <expr>] ...) <expr>)
+;;             | (if <expr> <expr> <expr>
+;;             | (cast <expr>)
+;;             | (! <props> ... <expr>)
+;;             | (<op> <expr> ...)
+;;             | <number>
+;;             | <id>
 ;;
-;; To propagate types through an FPCore, the types of the rounding context and
-;; variables must be known. Variable types propagate up and rounding contexts
-;; propagate down (modified only by !)
+;;  <var>    ::= <id>
+;;             | (! <props> ... <expr>)
 ;;
-;; ## ImplProg ##
+;; LImpl: the language of floating-point expressions
 ;;
-;; - internal language, floating-point programs
-;; - rounding context from FPCore embedded in operators (called implementations)
+;; - internal language, describes floating-point expressions
+;; - unary minus represented by `neg`
+;; - every operator is rounded
 ;; - let expressions are inlined
+;; - numbers are rounded to a particular format
 ;;
 ;; <expr> ::= (if <expr> <expr> <expr>)
 ;;        ::= (<impl> <expr> ...)
-;;        ::= <var>
-;;        ::= (literal <number> <repr-name>)
+;;        ::= (literal <number> <repr>)
+;;        ::= <id>
 ;;
-;; Every (operator) implementation has a type signature taking inputs each
-;; with a given number format producing an output of a possibly different
-;; number format. In practice, most operator implemenetations have uniform
-;; precision with only casts being multi-precision.
+;; Every operator has a type signature where types are representations.
+;; In practice, most operator implemenetations have uniform representations but
+;; operations like casts are multi-format.
 ;;
-;; ## Spec ##
+;; LSpec: the language of mathematical formula
 ;;
-;; - internal language, (almost) real-number programs
+;; - internal language, describes real-number formula
+;; - unary minus represented by `neg`
+;; - every operator is a real-number operation
+;; - every operator is atomic (cannot be decomposed into simpler operations)
```

Comment:
```suggestion
;; - every operator is supported by Rival
```

I know what you mean but "atomic" doesn't really make sense in this way.

Reply:
Fixed